src/IJulia.jl

```diff
@@ -211,6 +211,36 @@ a subset of cell inputs/outputs to clear.
 clear_history
 
 #######################################################################
+# methods to print history or any subset thereof
+function history(io::IO, indices)
+	# since a range could be huge, intersect it with 1:n first
+    for n in intersect(indices, 1:IJulia.n)
+      if haskey(In, n)
+        print(In[n])
+      end
+    end
+end
+history(x) = history(Base.STDOUT, x)
+history() = history(1:n)
+
+# print the last x history entries when argument x is integer
+history(io::IO, x::Integer) =
+    invoke(history, Tuple{IO, Any}, io, n-x:n)
```

Comment:
I thought it would be nice to have history(x::Integer) to show the latest x elements of IJulia.In. But this is of course different from how IPython's %history behaves. Maybe we should try to stay similar. 

In this spirit, we should, however, stick with `for n in intersect(indices, 1:IJulia.n)` and neither sort nor check for uniqueness, as `%history 4 4 2` in IPython is perfectly valid and gives two times the fourth and the second history entry.

I'll change the rest according to your nice suggestion.